udata/core/dataset/models.py

```diff
@@ -150,6 +149,68 @@ class HarvestResourceMetadata(DynamicEmbeddedDocument):
     uri = db.StringField()
 
 
+class Schema(db.EmbeddedDocument):
+    url = db.URLField()
+    name = db.StringField()
+    version = db.StringField()
+
+    def __bool__(self):
+        '''
+        In the database, since the schemas were only simple dicts, there is
+        empty `{}` stored. To prevent problems with converting to bool 
+        (bool({}) and default bool(Schema) do not yield the same value), we transform
+        empty Schema() to False.
+        It's maybe not necessary but being paranoid here.
+        '''
+        return bool(self.name) or bool(self.url)
+
+    def get_url(self):
```

Comment:
I think we don't use `get_name` and `get_url` anymore since we've simplified the api logic (returning what's stored only)?

Reply:
Yes, we can add them back if needed (for reference the commit of deletion 75585fc64)